### DISCUSSION TITLE
Octopus Intelligent Sensors Beta

### DISCUSSION THREAD

**Discussion by BottlecapDave** [BottlecapDave/HomeAssistant-OctopusEnergy#282] [Show and tell]
This is to discuss the beta for the Octopus Intelligent Sensors before it goes into the wild. I'll post here when new releases are available, and hopefully we can get this released properly in the future :)

Available sensors:
* switch.octopus_energy_intelligent_bump_charge - Enable/disable bump charges
* number.octopus_energy_intelligent_charge_limit - Set the target charge
* time.octopus_energy_intelligent_ready_time - Set the target ready time
* switch.octopus_energy_intelligent_smart_charge -Enable/disable smart charge
* binary_sensor.octopus_energy_intelligent_dispatching - See planned and completed dispatches and the sensor will be on when you're in one.

**Comment by BottlecapDave** [comment]
First release - https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/releases/tag/intelligent-sensors-beta1

> This release sees the introduction of new sensors available to Octopus Intelligent users. They include
    * switch.octopus_energy_intelligent_bump_charge - Enable/disable bump charges
    * number.octopus_energy_intelligent_charge_limit_weekday - Set the target charge for weekdays
    * number.octopus_energy_intelligent_charge_limit_weekend - Set the target charge for weekends
    * time.octopus_energy_intelligent_ready_time_weekday - Set the target ready time for weekdays
    * time.octopus_energy_intelligent_ready_time_weekend - Set the target ready time for weekends
    * switch.octopus_energy_intelligent_smart_charge -Enable/disable smart charge
    * binary_sensor.octopus_energy_intelligent_dispatching - See planned and completed dispatches and the sensor will be on when you're in one.

This is the first release and I've been obviously developing it a bit blind as I'm not on the OE tariff. Please report if you find any issues. Ongoing questions I have is

1. The API split up weekday/weekend targets but I'm not sure these are exposed via the app. If not, does this cause issues when moving between the app and HA. Is the granularity good? Should it match the app and just set both weekday/weekend to the same value (I believe this is how other integrations does it)
2. Does device information appear correctly? I've pointed it at the charger, but does it make more sense to point it at the car?

Thanks for the feedback

**Reply by megakid** [reply]
Good work. Will try it out shortly.

I made the decision in my integration to merge the weekend/weekday target soc and times since the app doesn't include the split and might cause confusion at best, or Octopus application issues at worst.

On the second point, I have only used it on my Tesla which interfaces directly with the *car* but with others such as Omhe it interfaces with the *charger*.

**Reply by BottlecapDave** [reply]
Released a new beta at https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/releases/tag/intelligent-sensors-beta2

> * Squashes weekday/weekend sensors into a single sensor to be in align with app and other integrations
> * Fixes unit of measurement for charge limit sensor
> * Fixes sensors handling errors reported by OE
> * Fixes refresh rate of OE intelligent settings

**Comment by RPCHome** [comment]
Hi @BottlecapDave,

I've just installed beta 2 and will see how it performs.

All the controls show up nicely. My only preference is to have a box to type in the Charge Limit % instead of the slider.

**Reply by BottlecapDave** [reply]
The sensor is set to `auto` mode which is what's recommended by HA. I assume this will be a profile setting in the future.

**Comment by BottlecapDave** [comment]
Any one got any additional feedback/bugs to report on these sensors?

**Comment by RPCHome** [comment]
They all seem to be working fine for me.

**Comment by RPCHome** [comment]
Just been having a play around with both you sensor control and @megakid sensor controls.

I know it's minor difference. Is there anyway to mirror yours to the same as @megakid ?

Beta
<img width="329" alt="Screenshot 2023-07-02 at 08 21 13" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/100522087/76ed51fd-db5b-4120-b7fb-130e6ffef27a">

@megakid 
<img width="331" alt="Screenshot 2023-07-02 at 08 20 57" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/100522087/6dc80c16-ef38-4207-8c2e-68b0e0c082e2">

My other half is finding it fiddley  to select the percentage using the slider vs the drop down box she has been using in the past.

**Reply by megakid** [reply]
I am also seeing smart charge on this integration the opposite to what it should. Eg when octopus app (and my integration) says smart charge is disabled, this integration says it's enabled and vice Verda

**Reply by BottlecapDave** [reply]
> I am also seeing smart charge on this integration the opposite to what it should

See the issue. I'll fix it when I get a chance.

> My other half is finding it fiddley to select the percentage using the slider vs the drop down box she has been using in the past.

I'm hestitant to change the type of sensor as it's the best fit for the data, especially with automations. However I believe the step/min/max for the sensor might be wrong compared to the app? If so, setting these might make the slider easier to use? If collectively people think that a box output would be better than a slider, then I can go against the HA recommendations?

**Reply by BottlecapDave** [reply]
The latest beta should fix the smart charge issue.

**Comment by RPCHome** [comment]
> The latest beta should fix the smart charge issue.

I can confirm that bata 4 fixed the smart charge issue

**Comment by BottlecapDave** [comment]
Hi everyone. Do we think this is good for release?

**Reply by RPCHome** [reply]
I can't see any issues with this beta and would say that is good for release.

**Reply by megakid** [reply]
Same here. Good to release I'd say. I'll probably retire my integration, recommending this one once it is out and people are happy

**Reply by RPCHome** [reply]
@megakid Thank you for creating your integration in the 1st place as without that we wouldn't be where we are today with @BottlecapDave integration.

Top work to the both of you.

**Reply by BottlecapDave** [reply]
This has now been released in the latest release (v7.5.0). Thanks for the help everyone!
